src/Authenticator/CookieAuthenticator.php

```diff
@@ -154,7 +156,17 @@ protected function _createPlainToken($identity): string
         $usernameField = $this->getConfig('fields.username');
         $passwordField = $this->getConfig('fields.password');
 
-        return $identity[$usernameField] . $identity[$passwordField];
+        if (!$this->getConfig('salt')) {
+            return $identity[$usernameField] . $identity[$passwordField];
+        }
+
+        if (is_string($this->getConfig('salt'))) {
+            $salt = $this->getConfig('salt');
+        } else {
+            $salt = Security::getSalt();
```

Comment:
this class needs to `cakephp/utility` dependency

Reply:
Good find. I've added it here: https://github.com/cakephp/authentication/pull/467/files#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34R14

That seem right?

Reply:
That's what I ment with "hard dependency". The package is currently a suggestion, and other authenticators make it optional.:

https://github.com/cakephp/authentication/blob/aafb6831bd3a9e9f3607ffb254db529d5f6cf998/src/Authenticator/HttpDigestAuthenticator.php#L63-L65

Reply:
I see, thanks @ndm2 . Updated the code and will squash commits next.

Reply:
The `utility` package is already a dependency through `http`. This isn't doing anything useful.

https://github.com/cakephp/http/blob/4.x/composer.json#L29

Reply:
True, but maybe that could change in the future. I'd say we keep it flexible as it is now (and as it was already for `HttpDigestAuthenticator` as @ndm2 pointed out).

Reply:
It isn't going to change. If we have a dependency, then it's valid to assume its use.

Reply:
Looks like the code in the other authenticators was added before the `utility` dependency was added to `http`. If it's an indirect dependency now, then it should indeed be fine, and the other authenticators could drop the optional handling too then.

Reply:
Right, I've changed it here. Should I change the same for HttpDigest in this PR?

Reply:
It seems this resolved.